snapcraft/internal/mangling.py

```diff
@@ -112,26 +114,22 @@ def handle_glibc_mismatch(*, elf_files: FrozenSet[elf.ElfFile],
                    'files and the linker version (2.23) used in the '
                    'base. These are the GLIBC versions required by '
                    'the primed files that do not match and will be '
-                   'patched:\n {}\n'
-                   'To work around this, the newer libc will be '
-                   'migrated into the snap, and these files will be '
-                   'patched to use it.'.format('\n'.join(formatted_list)))
+                   'patched:\n{}\n'.format('\n'.join(formatted_list)))
     # We assume the current system will satisfy the GLIBC requirement,
     # get the current libc6 libraries (which includes the linker)
-    libc6_libraries = repo.Repo.get_package_libraries('libc6')
-    libc6_path = os.path.join('snap', 'libc6')
+    libc6_libraries_list = repo.Repo.get_package_libraries('libc6')
 
-    # Before doing anything else, verify there's a dynamic linker we can use.
-    dynamic_linker_path = os.path.join(
-        snap_base_path, libc6_path, _get_dynamic_linker(libc6_libraries))
+    # For security reasons, we do not want to automatically pull in
+    # libraries but expect them to be consciously brought in by stage-packages
+    # instead.
+    libc6_libraries_paths = [os.path.join(root_path, l[1:])
+                             for l in libc6_libraries_list]
 
-    dest_dir = os.path.join(root_path, libc6_path)
-    os.makedirs(dest_dir, exist_ok=True)
+    dynamic_linker = _get_dynamic_linker(libc6_libraries_paths)
 
-    for src in libc6_libraries:
-        dst = os.path.join(dest_dir, os.path.basename(src))
-        # follow_symlinks is set to True for elf crawling to work.
-        file_utils.link_or_copy(src, dst, follow_symlinks=True)
+    # Before doing anything else, verify there's a dynamic linker we can use.
```

Comment:
This comment is misleading, because we are not verifying anything here, we are just joining a path.